test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java

```diff
@@ -2063,6 +2064,7 @@ protected Settings nodeSettings(int nodeOrdinal, Settings otherSettings) {
             builder.put(IndexingPressure.SPLIT_BULK_HIGH_WATERMARK.getKey(), randomFrom("1KB", "16KB", "64KB"));
             builder.put(IndexingPressure.SPLIT_BULK_HIGH_WATERMARK_SIZE.getKey(), "256B");
         }
+        builder.put(ThreadPoolMergeScheduler.USE_THREAD_POOL_MERGE_SCHEDULER_SETTING.getKey(), randomBoolean());
```

Comment:
:thu: though perhaps we want an ability to override for those few tests that rely on specific merge behavior. We'll see, but we should provoke extra test runs to be sure this is alright.

Reply:
When we enable this, can we then assert busy at the end of the test, checking a few things on the merge-queue? For instance that the active io throttled task count becomes 0 and the currently running tasks becomes empty.

Reply:
> :thu: though perhaps we want an ability to override for those few tests that rely on specific merge behavior.

That's always permitted, when overriding `ESIntegTestCase#nodeSettings`, like `InternalEngineMergeIT` does.

>When we enable this, can we then assert busy at the end of the test, checking a few things on the merge-queue? 

Addressed with `InternalTestCluster#assertMergeQueueIsEmpty` from 436ef52c (which unfortunately addresses some other thing as well) and also merged https://github.com/elastic/elasticsearch/pull/122639 in order to re-enable such assertions for test-scoped ITs.